src/mame/sega/suprloco.cpp

```diff
@@ -271,10 +336,38 @@ void suprloco_state::draw_sprite(bitmap_ind16 &bitmap, const rectangle &cliprect
 	}
 	else
 	{
-		adjy = sy + height - 1;  // some of the sprites are still off by a pixel
+		adjy = sy + 31 + height; // sprites are offset by 256-(224+1)=31 pixels
+								 //  in flipscreen mode due to the way the
+								 //  video timing pals and sprite line compare
+								 //  counter reload value works
 		dy = -1;
 	}
 
+	/*
+	Palette PROM address bits:
+	//--------- pallete bank bits from the PPI
+	||/-------- 0 for sprites, 1 for tilemap
+	|||///----- palette bank bits from the tilemap for the underlying tile
+	||||||////- currently selected color based on the 4 tilemap color bits or 4 sprite bits
+	9876543210
+	*/
+	pen_t const pen_base = ((BIT(m_control,5,2)<<8)
+							| 0 /* bit 7 will always be zero for sprites */
+							| ((adjy<128)?0x10:0x00) ///TODO: horrible hack, see below.
+							);
+	/* Horrible hack explanation: The 0x10 bit should be pulled, along with
+	    0x20 and 0x40, from the tilemap 'behind' the sprite.
+	   To actually implement this, we could be doing something like rendering
+	    the sprite's color prom address, instead of its actual color data, into
+	    a sprite bitmap. Then, when baking the bitmap and the tilemap together,
+	    the sprite colors, if they are to be displayed (and not masked by bg),
+	    need to use those bits from whatever tilemap pixel they are on top of!
+	   This hack makes it so this bit should be set on the top half of the
+	    screen and clear on the bottom half, regardless of flipscreen state
+	    (i.e. it will be on the bottom half when flipped).
+	   It's technically wrong, but looks correct.
+	*/
```

Comment:
Wouldn’t it be better to render the necessary data into intermediate indexed bitmaps and then do the mix with custom code?  Pop Bingo (dooyong/dooyong.cpp) does this for the effective 8-bit background that’s actually implemented as two 4-bit tilemaps.

Reply:
I'm looking into this; this may take some time to complete, though.